Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs

```diff
@@ -121,6 +117,34 @@ public AddinStatus[] GetPendingUninstalls ()
 		
 		public static DatabaseConfiguration Read (string file)
 		{
+			var config = ReadInternal (file);
+			// Try to read application level config to support disabling add-ins by default.
+			var appConfig = ReadAppConfig ();
+
+			if (appConfig == null)
+				return config;
+
+			// Overwrite app config values with user config values
+			foreach (var entry in config.addinStatus)
+				appConfig.addinStatus [entry.Key] = entry.Value;
+
+			return appConfig;
+		}
+		
+		public static DatabaseConfiguration ReadAppConfig()
+		{
+			var exePath = System.Reflection.Assembly.GetExecutingAssembly ().Location;
+			var exeDirectory = Path.GetDirectoryName (exePath);
+			var appAddinsConfigFilePath = Path.Combine (exeDirectory, "addins-config.xml");
+
+			if (!File.Exists (appAddinsConfigFilePath))
+				return new DatabaseConfiguration ();
```

Comment:
As you can see in AddinDatabase.cs above, the caller checks for the existence of the file before calling DatabaseConfiguration.Read (ConfigFile) and I assume there is handling of exceptions somewhere in the call-stack. I would prefer keep the logic as it was.